sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1860,4 +1861,193 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     val structField = StructField(identifier.getText, typedVisit(dataType), nullable = true)
     if (STRING == null) structField else structField.withComment(string(STRING))
   }
+
+  /**
```

Comment:
I didn't review this file carefully, assuming it's just copy-paste code from `SparkSqlAstBuilder`

Reply:
Yes, this is moving what is needed from `SparkSqlAstBuilder`. The only real changes are in the `visitCreateTable` method.

These rules should have already been in the abstract class. I'm not sure why they were in `SparkSqlAstBuilder` other than that was the easiest place to put them when they were added.